Finally add the test that hit all three bugs from the previous patches.

Index: twisted/internet/test/test_win32events.py
===================================================================
--- twisted/internet/test/test_win32events.py	(patch further-test-cleanup)
+++ twisted/internet/test/test_win32events.py	(patch test-manual-reset-event)
@@ -322,5 +322,26 @@
         self.assertSignaled(event)
 
 
+    def test_addManualResetEvent(self):
+        """
+        When a manual reset event which has been added to the reactor is set,
+        the action associated with the event is invoked but its state is
+        left unchanged.
+        """
+        reactor = self.buildReactor()
+        event = self.createEvent(manualReset=True, initialState=True)
+
+        def cleanup(ignored):
+            reactor.removeEvent(event)
+            reactor.stop()
+
+        finished = Deferred().addCallback(cleanup)
+        listener = Listener(finished)
+        reactor.addEvent(event, listener, 'occurred')
+        self.runReactor(reactor)
+        self.assertTrue(listener.success)
+        self.assertSignaled(event)
+
+
 
 globals().update(Win32EventsTestsBuilder.makeTestCaseClasses())
